### DISCUSSION TITLE
Having trouble running player-sd-audiokit.ino on ESP32 AudioKit V2.2

### DISCUSSION THREAD

**Discussion by jimjing** [pschatzmann/arduino-audio-tools#1486] [Q&A] [answered]
I got my board at [amazon](https://www.amazon.com/JESSINIE-Bluetooth-ESP32-A1S-Development-Dual-core/dp/B0B5XKG5NK).

I am able to run the streams-url_mp3-audiokit example nicely. But when I try to run the player-sd-audiokit.ino example, I am getting in the console repeated `[E] AudioSourceSD.h : 48 - SD.begin cs=13 failed` errors. Did the pin for chip select changed for this version of the board? Is there any easy way to find out?

I also confirmed the dip switches on the board are OFF, ON, ON, OFF, OFF

Thanks. This is a really well documented project.

**Comment by pschatzmann** [comment, marked as answer]
You can find the relevant information in my old AudioKit project:
https://github.com/pschatzmann/arduino-audiokit/wiki/Problems-with-the-SD-Drive-or-some-conflicting-Buttons

**Reply by jimjing** [reply]
Thanks for the reply. I have triple checked my dip switches and used continuity test to make sure the pins are connected correctly as well. What is IO13 conflicting with in this case? Do I just have a faulty board?

**Comment by pschatzmann** [comment]
Assuming you use Es8388v1, it would be the second key. I am getting this message as well and as long as you don't press any key the SD should still work, even if dip switch 1 is on, but to avoid any issues it is supposed to be switched of. Don't forget to reboot after changing the dip settings.

The SD pins are the same across all boards....

```C++
class PinsAudioKitEs8388v1Class : public DriverPins {
 public:
  PinsAudioKitEs8388v1Class() {
    // sd pins
    addSPI(ESP32PinsSD);
    // add i2c codec pins: scl, sda, port, frequency
    addI2C(PinFunction::CODEC, 32, 33);
    // add i2s pins: mclk, bck, ws,data_out, data_in ,(port)
    addI2S(PinFunction::CODEC, 0, 27, 25, 26, 35);

    // add other pins
    addPin(PinFunction::KEY, 36, PinLogic::InputActiveLow, 1);
    addPin(PinFunction::KEY, 13, PinLogic::InputActiveLow, 2);
    addPin(PinFunction::KEY, 19, PinLogic::InputActiveLow, 3);
    addPin(PinFunction::KEY, 23, PinLogic::InputActiveLow, 4);
    addPin(PinFunction::KEY, 18, PinLogic::InputActiveLow, 5);
    addPin(PinFunction::KEY, 5, PinLogic::InputActiveLow, 6);
    addPin(PinFunction::AUXIN_DETECT, 12, PinLogic::InputActiveLow);
    addPin(PinFunction::HEADPHONE_DETECT, 39, PinLogic::InputActiveLow);
    addPin(PinFunction::PA, 21, PinLogic::Output);
    addPin(PinFunction::LED, 22, PinLogic::Output);
  }
};
```

Try to comment out all action commands in the sketch and double check if that helps...

**Reply by jimjing** [reply]
Comment out all action commands unfortunately didn't help. Just to confirm, when you say
>I am getting this message as well

Are you saying you are getting repeated `[E] AudioSourceSD.h : 48 - SD.begin cs=13 failed` as well when you run the `player_sd_audiokit` example? Are you able to play any mp3 from the sd card in that case? 

I also saw at the beginning of the log `Warning: Pin '13' not set up because of conflict` which could explain the ` SD.begin cs=13 failed` error.

**Comment by pschatzmann** [comment]
I confirm that I am getting this message as well: it is there to inform you that the Pin definition for the button is ignored in favor of the SD setup.

I suggest that you try to run the SD example that comes with ESP32 Arduino by adding the following 
```C++
// These pins are defined in the HAL
#define PIN_AUDIO_KIT_SD_CARD_CS 13  
#define PIN_AUDIO_KIT_SD_CARD_MISO 2 
#define PIN_AUDIO_KIT_SD_CARD_MOSI 15
#define PIN_AUDIO_KIT_SD_CARD_CLK  14

void setup() {
    SPI.begin(PIN_AUDIO_KIT_SD_CARD_CLK, PIN_AUDIO_KIT_SD_CARD_MISO, PIN_AUDIO_KIT_SD_CARD_MOSI, PIN_AUDIO_KIT_SD_CARD_CS);
 ```

Don't forget to change SD.begin() to SD.begin(PIN_AUDIO_KIT_SD_CARD_CS) in the sketch

ps. did you try to change the pin switch and then set them back again to off, on, on, off, off ?
